src/core/hle/service/http/http_c.cpp

```diff
@@ -147,12 +147,28 @@ static void ReceiveData(Service::Interface* self) {
 
     std::lock_guard<std::mutex> lock(map_it->second->mutex);
     const std::vector<u8>& data = map_it->second->response_data;
-    if (buf_size < data.size()) {
+
+    //TODO: Check if this should use context->downloaded_size instead of data.size()
+    if (buf_size > data.size()) {
         cmd_buff[1] = 0xD840A02B;
         return;
     }
 
-    std::copy(data.begin(), data.end(), buffer);
+    if (buf_size < data.size()) {
+        long data_size;
+
+        if (map_it->second->current_content_length + buf_size > map_it->second->content_length)
+            data_size = (u32)map_it->second->content_length - map_it->second->current_content_length;
+        else
+            data_size = buf_size;
+
+        std::copy(data.begin() + map_it->second->current_content_length, data.begin() + map_it->second->current_content_length + data_size, buffer);
+        map_it->second->current_content_length += data_size;
+    }
+    else{
```

Comment:
Style: `else` on the same line as the closing bracket, and a space before the opening bracket.